Fix ResourceNotFoundError exception message

mallamanis opened the pull request:
From `None` to the `identity_pool_id`.

bblommers reviewed (changes requested):
Hi @mallamanis, thanks for the PR! Would you mind adding a unit test for this? Happy to merge after that

mallamanis commented:
@bblommers Done. I updated an existing test that was already there. Note that the test previously had an incorrect indentation for the fine-grained asserts of `pytest.raises`, which is also fixed now.

bblommers reviewed (approved):
LGTM - Thanks @mallamanis!
